src/util.py

```diff
@@ -260,12 +265,145 @@ def xr_add_cyclic_longitudes(da, coord):
+    """
+    Utility function to find local low/high field variable coordinates on a contour map
+
+    Args:
+
+        da: (:class:`xarray.DataArray`):
+            Xarray data array containing the lat, lon, and field variable (ex. pressure) data values
+
+        maxVal (:class:`int`):
+            Data value that the local maximum must be greater than
+            to qualify as a "local high" location
+
+        minVal (:class:`int`):
+            Data value that the local minimum must be less than
+            to qualify as a "local low" location
+
+        eType (:class:`str`):
+            'Min' or 'Max'
+            Determines which extrema are being found- minimum or maximum,
+            respectively
+    
+    Returns: 
+    
+        clusterExtremas (:class:`list`):
+            List of coordinate tuples in GPS form (lon in degrees, lat in degrees)
+            that specify local minimum/maximum locations
+            
+    """
+    import numpy as np
+    from sklearn.cluster import DBSCAN
+
+    # Create a 2D array of coordinates in the same shape as the field variable data
+    # so each coordinate is easily mappable to a data value
+    coordarr = []
+    for y in np.array(da.lat):
+        temparr = []
+        for x in np.array(da.lon):
+            temparr.append((x, y))
+        coordarr.append(temparr)
+    coordarr =  np.array(coordarr)
+
+    # Set number that a derivative must be less than in order to
+    # classify as a "zero"
+    bound = 0.0
+
+    # Get global gradient of contour data
+    grad = np.gradient(da.data)
+
+    # Gradient in the x direction
+    arr1 = grad[0]
+
+    # Gradient in the y direction
+    arr2 = grad[1]
+
+    # Get all array 1 indexes where gradient value is between -bound and +bound
+    posfirstzeroes = np.argwhere(arr1 <= bound)
+    negfirstzeroes = np.argwhere(-bound <= arr1)
+
+    # Get all array 2 indexes where gradient value is between -bound and +bound
+    possecondzeroes = np.argwhere(arr2 <= bound)
+    negsecondzeroes = np.argwhere(-bound <= arr2)
+
+    # Find zeroes of all four gradient arrays
+    commonzeroes = []
+    for x in possecondzeroes:
+        if x in posfirstzeroes and x in negfirstzeroes and x in negsecondzeroes:
+            commonzeroes.append(x)
+
+    # Find all zeroes that also qualify as minimum or maximum values
+    extremacoords = []
+    for x in commonzeroes:
+        try:
+            # xval is x index of the zero and yval is y index of the zero
+            xval = x[0]
+            yval = x[1]
+
+            # If the field variable value at the coordinate is less than minVal:
+            if eType == 'Min' and da.data[xval][yval] < minVal:
+                # Add coordinate as an extrema
+                extremacoords.append(tuple(coordarr[xval][yval]))
+            # If the field variable value at the coordinate is greater than maxVal:
+            if eType == 'Max' and da.data[xval][yval] > maxVal:
+                # Add coordinate as an extrema
```

Comment:
See if you can find a way to produce an error when there aren't any highs or lows that are greater than `maxVal` or less than `minVal`.  The program crashes when that happens and doesn't produce a helpful error message.